tests/php/Control/UserDefinedFormControllerTest.php

```diff
@@ -467,4 +468,31 @@ public function testRecipientAttachment()
         $nodata = $this->findEmail('nodata@example.com', 'no-reply@example.com', 'Email Subject');
         $this->assertEmpty($nodata['AttachedFiles'], 'Recipients with HideFormData do not receive attachment');
     }
+
+    public function testCreateNONExistedFolder()
+    {
+        Config::modify()->set(Upload_Validator::class, 'use_is_uploaded_file', false);
+        $userForm = $this->setupFormFrontend('upload-form-without-folder');
+        $controller = UserDefinedFormController::create($userForm);
+        $field = $this->objFromFixture(EditableFileField::class, 'file-field-3');
+
+        $path = realpath(__DIR__ . '/fixtures/testfile.jpg');
+        $data = [
+            $field->Name => [
+                'name' => 'testfile.jpg',
+                'type' => 'image/jpeg',
+                'tmp_name' => $path,
+                'error' => 0,
+                'size' => filesize($path ?? ''),
+            ]
+        ];
+        $_FILES[$field->Name] = $data[$field->Name];
+
+        $controller->getRequest()->setSession(new Session([]));
+
+        Versioned::set_stage(Versioned::LIVE);
+        $controller->process($data, $controller->Form());
+
+        $this->assertEquals(Versioned::LIVE, Versioned::get_stage());
```

Comment:
Why are we testing Versioned::get_stage() here?  This isn't related to the pull-request

Shouldn't we be asserting that
a) there's currently no folder before uploading the file
b) after uploading a file, a folder was created that the file is now it?